README.md

```diff
@@ -62,7 +69,65 @@ const exportedTypeSuite: t.ITypeSuite = {
 export default exportedTypeSuite;
 ```
 
-See [ts-interface-checker](https://github.com/gristlabs/ts-interface-checker) module for how to use this file in your program.
+See [ts-interface-checker documentation](https://github.com/gristlabs/ts-interface-checker#readme) for how to use the exported type suite in your code.
+
+## Babel Macro Usage
+
+**Call a Babel macro function that returns a generated type suite**
+
+This method requires your project to have [Babel](https://babeljs.io/docs/en/) and [`babel-plugin-macros`](https://github.com/kentcdodds/babel-plugin-macros) set up.
+See installation instructions for each in their respective documentation.
+Note that if you're using a Babel preset, you might unknowingly already have `babel-plugin-macros` set up.
+For example, the popular [`babel-preset-react-app`](https://github.com/facebook/create-react-app/tree/master/packages/babel-preset-react-app)
+already includes `babel-plugin-macros`.
+
+With those things set up, in your code you can now just call the `getTypeSuite` macro function (exported from `ts-interface-builder/macro`) to get a type suite.
+During your Babel build step, the required type suites will be generated and included in Babel's output.
+
+For example, if you have a TypeScript file that defines some types:
+```typescript
+// foo.ts
+interface Square {
+  size: number;
+  color?: string;
+}
+```
+
+You can get a type suite for it in your code like this:
+```typescript
+import { getTypeSuite } from 'ts-interface-builder/macro';
+
+const fooTypeSuite = getTypeSuite('./foo.ts', { /* options */ });
+```
+
+The code above would be transpiled into:
```

Comment:
I think I'd skip this block. It's long, the details are mostly internal. How about something more illustrative. Maybe mention  that the runtime description is generated from the interfaces found in `'./foo.ts'`, so that this becomes roughly equivalent to:
```
const fooTypeSuite = {
    Square: t.iface([], {
      "size": "number",
      "color": t.opt("string")
    })
  };
```